src/provider.ts

```diff
@@ -90,6 +91,28 @@ export class HyphenProvider implements Provider {
     hookContext.logger.info('logging usage');
   };
 
+  afterHook = async (hookContext: HookContext, evaluationDetails: EvaluationDetails<FlagValue>): Promise<void> => {
+    const parsedEvaluationDetails = {
+      key: evaluationDetails.flagKey,
+      value: evaluationDetails.value,
+      type: hookContext.flagValueType,
+      reason: evaluationDetails.reason,
+    };
+
+    try {
+      const payload: TelemetryPayload = {
+        context: hookContext.context as HyphenEvaluationContext,
+        data: { toggle: parsedEvaluationDetails as Evaluation},
+      };
+
+      await this.hyphenClient.postTelemetry(payload);
+      hookContext.logger.info('Payload sent to postTelemetry:', JSON.stringify(payload));
+    } catch (error) {
+      hookContext.logger.error('Error in afterHook:', error);
+      throw error;
+    }
+  };
+
```

Comment:
We should probably have a flag that can turn this off. an option in the constructor would be good enough.